package.json

```diff
@@ -2,8 +2,14 @@
   "name": "leaflet.functionaltilelayer",
   "version": "0.0.0",
   "description": "Leaflet tile layer with functionally defined URL and support for promises.",
-  "keywords": ["leaflet"],
-  "main": "src/leaflet.functionaltilelayer.js",
+  "keywords": [
+    "leaflet"
+  ],
+  "main": "./src/leaflet.functionaltilelayer.js",
+  "types": "./src/leaflet.functionaltilelayer.d.ts",
   "author": "Ishmael Smyrnow",
-  "license": "MIT"
+  "license": "MIT",
+  "dependencies": {
+    "@types/leaflet": "^1.0.34"
```

Comment:
The TypeScript-Handbook suggests to use `dependencies`: [Publishing](http://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html). But i understand your concerns. I will remove the TypeScript-stuff and maintain this in a fork. I will read about it and maybe create a different pull request.